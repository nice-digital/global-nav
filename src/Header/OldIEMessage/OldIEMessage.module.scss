@use '@nice-digital/nds-core/scss/colours';
@use '@nice-digital/nds-core/scss/layout';
@use '@nice-digital/nds-core/scss/spacing';
@use '@nice-digital/nds-core/scss/typography';
@use '@nice-digital/nds-core/scss/utils';

.container {
  @include layout.container;
  margin-bottom: utils.rem(spacing.$medium);
}

.alert {
  background: colours.$caution-background;
  border: 1px solid colours.$caution;
  margin: utils.rem(spacing.$medium) auto 0 auto;
  max-width: utils.rem(layout.$container-max-width);
  padding: utils.rem(spacing.$medium);

  a {
    @include typography.link-default;
  }

  h2 {
    margin-top: 0;
  }

  h2,
  p {
    max-width: 600px;
  }
}

.lastP {
  margin-bottom: 0;
}
