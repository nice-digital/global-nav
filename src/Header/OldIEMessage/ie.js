// See https://gist.github.com/padolsey/527683#gistcomment-768383
export const getIEVersion = () => {
	if (typeof document === "undefined") return void 0;
	let tmp = (document["documentMode"] || document.attachEvent) && "ev",
		msie =
			tmp &&
			(tmp = window[tmp + "al"]) &&
			tmp("/*@cc_on 1;@*/") &&
			+((/msie (\d+)/i.exec(navigator.userAgent) || [])[1] || 0);
	return msie || void 0;
};
