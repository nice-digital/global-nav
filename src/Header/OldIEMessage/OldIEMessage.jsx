import React from "react";
import { getIEVersion } from "./ie";

import styles from "./OldIEMessage.module.scss";

export default function OldIEMessage() {
	const ieVersion = getIEVersion();

	if (!ieVersion || ieVersion > 10) {
		return null;
	}

	return (
		<aside className={styles.container}>
			<div className={styles.alert}>
				<h2>Browser support</h2>
				{ieVersion === 10 ? (
					<div>
						<p>
							Your browser, Internet Explorer (
							<abbr title="Internet Explorer">IE</abbr>) 10, will reach end of
							support on{" "}
							<strong>
								<time dateTime="2020-01-31">January 31, 2020</time>
							</strong>
							. Microsoft will no longer issue security patches from this date
							and your browser may become insecure. You can learn more about
							this in Microsoftâ€™s{" "}
							<a href="https://support.microsoft.com/en-gb/help/17454/lifecycle-faq-internet-explorer">
								Internet Explorer Lifecycle Policy FAQ
							</a>
							.
						</p>
						<p>
							NICE will also stop supporting{" "}
							<abbr title="Internet Explorer">IE</abbr> 10 from{" "}
							<time dateTime="2020-01-31">January 31, 2020</time>. This means
							NICE websites and services may stop functioning correctly in{" "}
							<abbr title="Internet Explorer">IE</abbr> 10 after this date.
						</p>
					</div>
				) : (
					<div>
						<p>
							Your web browser, Internet Explorer (
							<abbr title="Internet Explorer">IE</abbr>) {ieVersion}, is out of
							date. For the best experience, please update your browser to the
							latest version.
						</p>
					</div>
				)}
				<p>
					We recommend you upgrade to a newer browser such as{" "}
					<a href="https://technet.microsoft.com/en-us/microsoft-edge">
						Microsoft Edge
					</a>
					, <a href="https://www.google.com/chrome/">Google Chrome</a> or&nbsp;
					<a href="https://www.mozilla.org/en-GB/firefox/new/">Firefox</a>.
				</p>
			</div>
		</aside>
	);
}
