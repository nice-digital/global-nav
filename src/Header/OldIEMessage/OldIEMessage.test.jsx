import React from "react";
import OldIEMessage from "./OldIEMessage";
import { render } from "@testing-library/react";

describe("OldIEMessage", () => {
	it("Doesn't render in IE11", () => {
		const { container } = render(<OldIEMessage />);

		wrapper.instance().getIEVersion = jest.fn(() => 11);
		wrapper.instance().forceUpdate();

		expect(wrapper.get(0)).toBeNull();
	});

	it("Doesn't render in modern browsers", () => {
		const { container } = render(<OldIEMessage />);

		wrapper.instance().getIEVersion = jest.fn(() => null);
		wrapper.instance().forceUpdate();

		expect(wrapper.get(0)).toBeNull();
	});

	it("Matches snapshot in IE 10", () => {
		const { container } = render(<OldIEMessage />);

		wrapper.instance().getIEVersion = jest.fn(() => 10);
		wrapper.instance().forceUpdate();

		expect(container).toMatchSnapshot();
	});

	it("Matches snapshot in IE 8/9", () => {
		const { container } = render(<OldIEMessage />);

		wrapper.instance().getIEVersion = jest.fn(() => 9);
		wrapper.instance().forceUpdate();

		expect(container).toMatchSnapshot();
	});
});
