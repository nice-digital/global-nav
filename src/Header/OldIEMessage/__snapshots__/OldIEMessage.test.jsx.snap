// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OldIEMessage Matches snapshot in IE 8/9 1`] = `
<div>
  <aside
    class="container"
  >
    <div
      class="alert"
    >
      <h2>
        Browser support
      </h2>
      <div>
        <p>
          Your web browser, Internet Explorer (
          <abbr
            title="Internet Explorer"
          >
            IE
          </abbr>
          ) 
          9
          , is out of date. For the best experience, please update your browser to the latest version.
        </p>
      </div>
      <p>
        We recommend you upgrade to a newer browser such as
         
        <a
          href="https://technet.microsoft.com/en-us/microsoft-edge"
        >
          Microsoft Edge
        </a>
        , 
        <a
          href="https://www.google.com/chrome/"
        >
          Google Chrome
        </a>
         or 
        <a
          href="https://www.mozilla.org/en-GB/firefox/new/"
        >
          Firefox
        </a>
        .
      </p>
    </div>
  </aside>
</div>
`;

exports[`OldIEMessage Matches snapshot in IE 10 1`] = `
<div>
  <aside
    class="container"
  >
    <div
      class="alert"
    >
      <h2>
        Browser support
      </h2>
      <div>
        <p>
          Your browser, Internet Explorer (
          <abbr
            title="Internet Explorer"
          >
            IE
          </abbr>
          ) 10, will reach end of support on
           
          <strong>
            <time
              datetime="2020-01-31"
            >
              January 31, 2020
            </time>
          </strong>
          . Microsoft will no longer issue security patches from this date and your browser may become insecure. You can learn more about this in Microsoft’s
           
          <a
            href="https://support.microsoft.com/en-gb/help/17454/lifecycle-faq-internet-explorer"
          >
            Internet Explorer Lifecycle Policy FAQ
          </a>
          .
        </p>
        <p>
          NICE will also stop supporting
           
          <abbr
            title="Internet Explorer"
          >
            IE
          </abbr>
           10 from
           
          <time
            datetime="2020-01-31"
          >
            January 31, 2020
          </time>
          . This means NICE websites and services may stop functioning correctly in
           
          <abbr
            title="Internet Explorer"
          >
            IE
          </abbr>
           10 after this date.
        </p>
      </div>
      <p>
        We recommend you upgrade to a newer browser such as
         
        <a
          href="https://technet.microsoft.com/en-us/microsoft-edge"
        >
          Microsoft Edge
        </a>
        , 
        <a
          href="https://www.google.com/chrome/"
        >
          Google Chrome
        </a>
         or 
        <a
          href="https://www.mozilla.org/en-GB/firefox/new/"
        >
          Firefox
        </a>
        .
      </p>
    </div>
  </aside>
</div>
`;
